### ISSUE TITLE
Environments or environment variables in workspaces?

### ISSUE THREAD

**Issue by bl-ue** [microsoft/vscode#130493]
It would be really cool to have some sort of feature, I'm calling it envirorments.

There could be lots of stuff associated with it, but in this case I've got a really simple problem: often I open VS Code from a terminal, because I need to run `export PATH=...` in order to be able to use some extension inside of VS Code that uses a binary to work, e.g., [this Lilypond extension](https://marketplace.visualstudio.com/items?itemName=lhl2617.VSLilyPond), which works really well (thank you @lhl2617!), uses the `lilypond` external binary.

I'd really like to be able to open a new VS Code windo, and then click some sort of link that opens a folder like normal, only with extra environment variables sourced from a user-defined environment.

Perhaps a workspace could have environment variables in addition to project folders? :thinking:

**Comment by lhl2617** [comment]
FYI that you don't need to shove the `PATH` there :) you can specify the PATH in one of the config.

https://github.com/lhl2617/VSLilyPond/blob/HEAD/docs/INSTALL.md#lilypond-setup-guide

**Comment by lhl2617** [comment]
Otherwise for workarounds:

On the extreme end, you can have a devcontainer that you specify env vars. This is likely overkill.

An easier solution is to use something like @EthanSK 's https://github.com/EthanSK/restore-terminals-vscode where you can just make your workspace run a command to set env vars.

**Comment by bl-ue** [comment]
Aha! Very nice trick. Dumb of me not to read such a simple guide :laughing:

Still, the theme of my idea remains the same.

**Comment by wickerwaka** [comment]
Yes, I agree that the idea is useful even if you specific example can be solved in another way. There are plenty of other cases where being able to set specific environment variables for a workspace would be useful.

**Comment by sanmai-NL** [comment]
This is important to set universally relevant configurations in a DRY way, such as `JAVA_HOME`, which is depended on by multiple popular VS Code extensions.
